typescript/core/src/v0/Mcap0StreamReader.ts

```diff
@@ -258,3 +263,24 @@ export default class Mcap0StreamReader implements McapStreamReader {
     }
   }
 }
+
+function isChannelEqual(a: Channel, b: Channel): boolean {
+  if (
+    !(
+      a.id === b.id &&
+      a.messageEncoding === b.messageEncoding &&
+      a.schemaId === b.schemaId &&
+      a.topic === b.topic &&
+      a.metadata.size === b.metadata.size
```

Comment:
Is there anything we can do to protect against adding fields but forgetting to define them here for equality comparison?

Random thought: might it be better to store a crc of the original record bytes and compare that?